src/main/java/johnny/Parser.java

```diff
@@ -0,0 +1,111 @@
+package johnny;
+
+import java.util.ArrayList;
+
+public class Parser {
+
+    private String input;
+
+    public static String TERMINATE = "TERMINATE";
+    public static String PRINT_LIST = "PRINT_LIST";
+    public static String MARK = "MARK";
+    public static String UNMARK = "UNMARK";
+    public static String DELETE = "DELETE";
+    public static String ADD_TODO = "ADD_TODO";
+    public static String ADD_DEADLINE = "ADD_DEADLINE";
+    public static String ADD_EVENT = "ADD_EVENT";
+    public static String FIND_EVENT = "FIND_EVENT";
+
+
+    public Parser(String userInput) {
+        this.input = userInput;
+    }
+
+    private static boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch(NumberFormatException e){
+            return false;
+        }
+    }
+
+    /**
+     * Parses input String into an Arraylist
+     * of tags depending on command
+     *
+     * @return Arraylist of tags.
+     * @throws EmptyDescriptionException  If description for tasks is not mentioned.
+     * @throws NoDateException  If no/incorrectly formatted date is entered for tasks.
+     * @throws InvalidArgumentsException  If input command is invalid.
+     */
+    public ArrayList<String> parse() throws EmptyDescriptionException, NoDateException, InvalidArgumentsException {
+        String[] tags = input.split(" ", 2);
+        ArrayList<String> result = new ArrayList<>();
```

Comment:
```suggestion
        ArrayList<String> results = new ArrayList<>();
```
Great job in using an ArrayList to store the results here! However, I think plural form should be used on names representing a collection of objects.